### PR TITLE
upgrade: To Vonk to-be-4.0.0 and SDK 2.0.2

### PR DESCRIPTION
**Pull request by cknaap**

### CODE REVIEW

**Comment by cknaap** [comment]
Build does not work as I built it against temporary packages from the vonk fix/vonk-2032-obsolete-api-4.0.0 branch. 
But if you could check the code changes, I will upgrade it to the proper 4.0.0 packages when we released them.

File: Facade.props [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <FirelySDKVersion>2.0.2</FirelySDKVersion>
```
Comment (alexzautke) [inline Facade.props:4]: Should be updated to 2.0.3 for consistency

**Review by alexzautke — APPROVED** [review]

**Review by vadi2 — APPROVED** [review]

File: Visi.Repository/VisiQueryFactory.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System;
+using Vonk.Core.Common;
+using Vonk.Facade.Relational;
+
+namespace Visi.Repository
+{
+    public abstract class VisiQueryFactory<E, Q> : RelationalQueryFactory<E, Q> where E : class where Q : RelationalQuery<E>, new()
```
Comment (vadi2) [inline Visi.Repository/VisiQueryFactory.cs:8]: I have to admit this is black magic

Comment (cknaap) [reply Visi.Repository/VisiQueryFactory.cs:8]: @vadi2 : Generics are from the dark side. Powerful but dangerous :-)

**Pull request merged**